code/espurna/ota.ino

```diff
@@ -55,107 +69,123 @@ void _otaFrom(const char * host, unsigned int port, const char * url) {
     _ota_port = port;
     _ota_size = 0;
 
-    if (_ota_client == NULL) {
-        _ota_client = new AsyncClient();
-    }
+    bool connected;
 
-    _ota_client->onDisconnect([](void *s, AsyncClient *c) {
+    DEBUG_MSG_P(PSTR("[OTA] Connecting to %s:%u\n"), host, port);
 
-        DEBUG_MSG_P(PSTR("\n"));
-
-        if (Update.end(true)){
-            DEBUG_MSG_P(PSTR("[OTA] Success: %u bytes\n"), _ota_size);
-            deferredReset(100, CUSTOM_RESET_OTA);
+    #if ASYNC_TCP_SSL_ENABLED
+        if (port == 443) {
+            WiFiClientSecure _ota_client = _ota_client_secure;
+            connected = _ota_client.connect(host, port);
+            if (connected) {
+                char fp[60] = {0};
+                if (sslFingerPrintChar(getSetting("otafp", OTA_GITHUB_FP).c_str(), fp)) {
+                    #ifdef USING_AXTLS
+                    if (!_ota_client.verify(fp, host)) {
+                    #else
+                    if (!_ota_client.setFingerprint(fp)) {
```

Comment:
That means it is bearssl, so we need to sed fingerprint before connecting.
Can't check rn, but doesn't it also allow to pass char'p directly?

Reply:
Oops, missed that one. I have fixed it now.
We can also pass in the fingerprint char* directly but it wouldn't add anything.